lib/web_ui/dev/screenshot_manager.dart

```diff
@@ -0,0 +1,224 @@
+      'deviceScaleFactor': 1,
+      'mobile': false,
+    });
+    final wip.WipResponse response = await wipConnection.sendCommand(
+        'Page.captureScreenshot', captureScreenshotParameters);
+
+    final Image screenshot =
+        decodePng(base64.decode(response.result['data'] as String));
+
+    return screenshot;
+  }
+}
+
+/// [ScreenshotManager] implementation for Safari.
+///
+/// This manager is will only be created/used for macOS.
+class IOSSafariScreenshotManager extends ScreenshotManager {
+
+  String get filenameSuffix => '.iOS_Safari';
+
+  IOSSafariScreenshotManager() {
+    final YamlMap browserLock = BrowserLock.instance.configuration;
+    _heightOfHeader = browserLock['ios-safari']['heightOfHeader'] as int;
+    _heightOfFooter = browserLock['ios-safari']['heightOfFooter'] as int;
+
+    /// Create the directory to use for taking screenshots, if it does not
+    /// exists.
+    if (!environment.webUiSimulatorScreenshotsDirectory.existsSync()) {
+      environment.webUiSimulatorScreenshotsDirectory.createSync();
+    }
+    // Temporary directories are deleted in the clenaup phase of after `felt`
+    // runs the tests.
+    temporaryDirectories.add(environment.webUiSimulatorScreenshotsDirectory);
+  }
+
+  /// Height of the part to crop from the top of the image.
+  ///
+  /// `xcrun simctl` command takes the screenshot of the entire simulator. We
+  /// are cropping top bit from screenshot, otherwise due to the clock on top of
+  /// the screen, the screenshot will differ between each run.
+  /// Note that this gap can change per phone and per iOS version. For more
+  /// details refer to `browser_lock.yaml` file.
+  int _heightOfHeader;
+
+  /// Height of the part to crop from the bottom of the image.
+  ///
+  /// This area is the footer navigation bar of the phone, it is not the area
+  /// used by tests (which is inside the browser).
+  int _heightOfFooter;
+
+  /// Used as a suffix for the temporary file names used for screenshots.
+  int _fileNameCounter = 0;
+
+  /// Capture a screenshot of entire simulator.
+  ///
+  /// Example screenshot with dimensions: W x H.
+  ///
+  ///  <----------  W ------------->
+  ///  _____________________________
+  /// | Phone Top bar (clock etc.)  |   Ʌ
+  /// |_____________________________|   |
+  /// | Broswer search bar          |   |
+  /// |_____________________________|   |
+  /// | Web page content            |   |
+  /// |                             |   |
+  /// |                             |
+  /// |                             |   H
+  /// |                             |
+  /// |                             |   |
+  /// |                             |   |
+  /// |                             |   |
+  /// |                             |   |
+  /// |_____________________________|   |
+  /// | Phone footer bar            |   |
+  /// |_____________________________|   V
+  ///
+  /// After taking the screenshot top and bottom parts of the image by
```

Comment:
nit: I think it's missing something, perhaps "...parts of the image **are cropped** by..."

Reply:
done